src/coreComponents/physicsSolvers/SolverBase.hpp

```diff
@@ -455,6 +482,32 @@ class SolverBase : public ExecutableGroup
                        real64 const scalingFactor,
                        DomainPartition & domain );
 
+  /**
+   * @brief updates the configuration (if needed) based on the state after a converged Newton loop.
+   * @param domain the domain containing the mesh and fields
+   * @return a bool that states whether the configuration used to solve the nonlinear loop is still valid or not.
+   */
+  virtual bool updateConfiguration( DomainPartition & domain );
+
+  /**
+   * @brief
+   * @param domain the domain containing the mesh and fields
+   */
+  virtual void outputConfigurationStatistics( DomainPartition const & domain ) const;
+
+  /**
+   * @brief resets the configuration to the beginning of the time-step.
+   * @param domain the domain containing the mesh and fields
+   */
+  virtual void resetConfigurationToBeginningOfStep( DomainPartition & domain );
+
+  /**
+   * @brief set the simplest configuration state.
+   * @param domain the domain containing the mesh and fields
+   */
+  virtual bool setSimplestConfigurationState( DomainPartition & domain ) const;
```

Comment:
Does this really make sense to all the solvers that inherit from `SolverBase`.
If not, I would argue that this should go down to `ContactSolverBase`, and replaced by a wide `simplify` (or any better name), see other comment.

Reply:
I think `defaultConfiguration` works fine.